plone/schemaeditor/browser/schema/schema_listing.pt

```diff
@@ -74,14 +74,16 @@
     <metal:fields-slot metal:fill-slot="fields">
 
       <tal:block tal:repeat="group python:[view] + list(view.groups)">
-        <fieldset tal:define="fieldset_name repeat/group/index"
+        <fieldset tal:define="fieldset_name repeat/group/index;
+                              can_delete python:view.can_delete_fieldset(group);"
                   tal:attributes="id string:fieldset-${fieldset_name};
-                                  class string:kssattr-fieldset-${fieldset_name}">
+                                  class string:kssattr-fieldset-${fieldset_name};
+                                  data-can-add-fields python:(repeat['group'].start or can_delete) and 'true' or 'false'">
 
           <legend tal:define="group_name python:group.label or view.default_fieldset_label"
                   tal:content="group_name">Fieldset name</legend>
 
-          <a id="delete-fieldset-${fieldset_name}" tal:condition="python:context.enableFieldsets and group.__name__ != view.__name__"
+          <a id="delete-fieldset-${fieldset_name}" tal:condition="python:context.enableFieldsets and can_delete"
```

Comment:
I would recommend to add the following bootstrap classes here: `btn btn-sm btn-danger`